File: Cpp/Maximum_Number_of_Operations_to_Move_Ones_to_the_End.cpp
class Solution {
public:
    int maxOperations(string s) {
        int op = 0;
        int once = 0;
        for (int i = 0; i < s.size(); i++) {
            if (s[i] == '1') {
                once++;
            } else if (i > 0 && s[i - 1] != '0') {
                op += once;
            }
        }
        return op;
    }
};

File: Rust/Maximum_Number_of_Operations_to_Move_Ones_to_the_End.rs
impl Solution {
    pub fn max_operations(s: String) -> i32 {
        let mut once = 0;
        let mut op = 0;
        let b = s.as_bytes();
        for i in 0..b.len() {
            if b[i] == b'1' {
                once += 1;
            } else if i > 0 && b[i - 1] != b'0' {
                op += once;
            }
        }
        op
    }
}
